src/util/color.js

```diff
@@ -99,28 +99,54 @@ Color.hexToDecimal = function (hex) {
  * @returns {RGBObject} rgb - {r: red [0,255], g: green [0,255], b: blue [0,255]}.
  */
 Color.hsvToRgb = function (hsv) {
-    var h = hsv.h;
-    var s = hsv.s;
-    var v = hsv.v;
-
-    h = h % 360;
+    var h = hsv.h % 360;
     if (h < 0) h += 360;
-    s = Math.max(0, Math.min(s, 1));
-    v = Math.max(0, Math.min(v, 1));
+    var s = Math.max(0, Math.min(hsv.s, 1));
+    var v = Math.max(0, Math.min(hsv.v, 1));
 
     var i = Math.floor(h / 60);
     var f = (h / 60) - i;
     var p = v * (1 - s);
     var q = v * (1 - (s * f));
     var t = v * (1 - (s * (1 - f)));
 
-    var r, g, b;
-    if (i == 0) { r = v; g = t; b = p; }
-    else if (i == 1) { r = q; g = v; b = p; }
-    else if (i == 2) { r = p; g = v; b = t; }
-    else if (i == 3) { r = p; g = q; b = v; }
-    else if (i == 4) { r = t; g = p; b = v; }
-    else if (i == 5) { r = v; g = p; b = q; }
+    var r;
+    var g;
+    var b;
+
+    switch (i) {
+    default:
+    case 0:
+        r = v;
+        g = t;
+        b = p;
+        break;
+    case 1:
+        r = q;
+        g = v;
+        b = p;
+        break;
+    case 2:
+        r = p;
+        g = v;
+        b = t;
+        break;
+    case 3:
+        r = p;
+        g = q;
+        b = v;
+        break;
+    case 4:
+        r = t;
+        g = p;
+        b = v;
+        break;
+    case 5:
+        r = v;
+        g = p;
+        b = q;
+        break;
+    }
```

Comment:
Hmph. This is (arguably) less readable than before, part because everything's on separate lines, somewhat away from the `case`s, and part because it's a switch-case, which gets you all the cludge of `case` and `break`...

Nitpicks :)